src/utils/resources/vm/hooks/useVMI.ts

```diff
@@ -1,20 +1,22 @@
 import { VirtualMachineInstanceModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
-import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import useK8sWatchData from '@kubevirt-utils/hooks/useK8sWatchData';
 
 type UseVMI = (
   vmName: string,
   vmNamespace: string,
+  vmCluster?: string,
   fetch?: boolean,
 ) => {
   vmi: V1VirtualMachineInstance;
   vmiLoaded: boolean;
   vmiLoadError: Error;
 };
 
-const useVMI: UseVMI = (vmName, vmNamespace, fetch = true) => {
-  const [vmi, vmiLoaded, vmiLoadError] = useK8sWatchResource<V1VirtualMachineInstance>(
+const useVMI: UseVMI = (vmName, vmNamespace, vmCluster, fetch = true) => {
```

Comment:
Can't we use the `useParams` hook inside the `useVMI` and `useVMIAndPodsForVMValues` hooks to avoid passing in the cluster? That would reduce the number of places the code needs to be modified.